content/blog/workflowsets-0-0-1/index.Rmd

```diff
@@ -0,0 +1,316 @@
+
+```{r map, results = "hide", message = FALSE}
+meat_wflow_set <-
+  meat_wflow_set %>%
+  workflow_map(
+    # Options to `tune_grid()`
+    resamples = meat_folds,
+    grid = 20,
+    metrics = metric_set(rmse),
+    # Options to `workflow_map()`
+    seed = 3,
+    verbose = TRUE
+  )
+meat_wflow_set
+```
+```
+i	No tuning parameters. `fit_resamples()` will be attempted
+i  1 of 12 resampling: plain_lm
+✓  1 of 12 resampling: plain_lm (2.3s)
+i	No tuning parameters. `fit_resamples()` will be attempted
+i  2 of 12 resampling: plain_mars
+✓  2 of 12 resampling: plain_mars (2.6s)
+i  3 of 12 tuning:     pca_lm
+✓  3 of 12 tuning:     pca_lm (8.1s)
+i  4 of 12 tuning:     pca_mars
+✓  4 of 12 tuning:     pca_mars (8.4s)
+i  5 of 12 tuning:     pca_rbf_lm
+✓  5 of 12 tuning:     pca_rbf_lm (36.9s)
+i  6 of 12 tuning:     pca_rbf_mars
+✓  6 of 12 tuning:     pca_rbf_mars (34.1s)
+i  7 of 12 tuning:     pls_lm
+✓  7 of 12 tuning:     pls_lm (8.6s)
+i  8 of 12 tuning:     pls_mars
+✓  8 of 12 tuning:     pls_mars (8.6s)
+i  9 of 12 tuning:     ica_lm
+✓  9 of 12 tuning:     ica_lm (8.4s)
+i 10 of 12 tuning:     ica_mars
+✓ 10 of 12 tuning:     ica_mars (8.4s)
+i 11 of 12 tuning:     mds_lm
+✓ 11 of 12 tuning:     mds_lm (39.3s)
+i 12 of 12 tuning:     mds_mars
+✓ 12 of 12 tuning:     mds_mars (40s)
+```
+
+
+The `result` column contains the results of `tune_grid()` (or `fit_resamples()` for the "plain" models). 
+
+There are a few convenience functions for ranking the results. To get the rankings of the models (and their tuning parameter sub-models) as a data frame: 
+
+```{r rank}
+rank_results(meat_wflow_set) %>% 
+  # To fit on the page:
+  select(-.metric, -std_err, -n) %>% 
+  relocate(rank, mean)
+```
+
+or as a plot:
+
+```{r plot-rank, fig.height=4.5, fig.width=8, out.width="100%", dev="svg"}
+autoplot(meat_wflow_set)
+```
+ 
+This shows the results for all tuning parameter combinations for each model. To get the best results _per model_:
+
+```{r plot-best-rank, fig.height=4.5, fig.width=8, out.width="100%", dev="svg"}
+autoplot(meat_wflow_set, select_best = TRUE)
+```
+
+It seems like the MARS model is doing slightly better than ordinary linear regression. The original predictors do well in terms of the mean RMSE. However, the correlation between predictors can cause instability in the results and this is reflected in the larger error bars for those models; reducing the dimensions may still be a good idea. 
+
+Once you decide on a model to keep, you can pull the workflow (or the tuning results) using [`pull_workflow()`](https://workflowsets.tidymodels.org/reference/pull_workflow_set_result.html) (or `pull_workflow_set_result()`). This can be finalized and fit on the training set as one normally would. 
```

Comment:
```suggestion
Once you decide on a model to keep, you can pull the workflow (or the tuning results) using [`pull_workflow()`](https://workflowsets.tidymodels.org/reference/pull_workflow_set_result.html) (or `pull_workflow_set_result()`). This can be [finalized and fit on the training set as one normally would](https://www.tmwr.org/grid-search.html#finalizing-the-model). 
```